### PR TITLE
Graphing Fix

### PR DESCRIPTION
**Pull request by wpride**
Fixes https://manage.dimagi.com/default.asp?248652 and another bug where bar graphs with no entries would not appear

@orangejenny one proposed solution

### CODE REVIEW

File: app/src/org/commcare/suite/model/graph/Graph.java [hunk 1]
```diff
@@ -173,7 +174,10 @@ private void evaluateSeries(GraphData graphData, EvaluationContext context) {
                     }
                     String x = s.evaluateX(refContext);
                     String y = s.evaluateY(refContext);
-                    if (x != null && y != null) {
+                    if (x != null && graphData.getType().equals(GraphUtil.TYPE_BAR)) {
```
Comment (orangejenny) [inline app/src/org/commcare/suite/model/graph/Graph.java]: This looks like it'll never be true for xy graphs, and therefore never draw any points on xy graphs. I'd expect it to be `x != null && (y == null || graphData.getType().equals(GraphUtil.TYPE_BAR))`

File: app/src/org/commcare/suite/model/graph/XYSeries.java [hunk 2]
```diff
@@ -76,23 +76,35 @@ public void setY(String y) {
...
 
+    public void removeConfiguration(String key) {
+        mConfiguration.remove(key);
+    }
+
     @Override
     public void setConfiguration(String key, Text value) {
-        mConfiguration.put(key, value);
+        if (mPointConfiguration.keySet().contains(key)) {
+            mPointConfiguration.remove(key);
```
Comment (orangejenny) [inline app/src/org/commcare/suite/model/graph/XYSeries.java]: Minor, but I think this is redundant; the `put` below will overwrite the current value.

File: app/src/org/commcare/suite/model/graph/Graph.java [hunk 3]
```diff
@@ -149,9 +149,10 @@ private void evaluateSeries(GraphData graphData, EvaluationContext context) {
                 Hashtable<String, Text> pointConfiguration = new Hashtable<>();
                 for (Enumeration e = s.getPointConfigurationKeys(); e.hasMoreElements();) {
                     String key = (String) e.nextElement();
-                    Text value = s.getConfiguration(key);
+                    Text value = s.getPointConfiguration(key);
                     if (value != null) {
                         pointConfiguration.put(key, value);
+                        s.removeConfiguration(key);
```
Comment (orangejenny) [inline app/src/org/commcare/suite/model/graph/Graph.java]: Not quite following why the key needs to be removed from the configuration...?

**Comment by amstone326** [comment]
@wpride Are you trying to get this into 2.35?

**Comment by wpride** [comment]
@orangejenny comments addressed, would be good to get this into 2.35

**Comment by orangejenny** [comment]
Fixed conditional because I made the wrong suggestion. Good to go.

**Pull request merged**